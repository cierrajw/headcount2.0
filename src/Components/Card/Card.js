import React from 'react';

const Card = ({location, stats}) =>{
  const statsArray = Object.keys(stats);
  const districtData = statsArray.map(data=>stats[data])
  return(
    <div>
    <h4>{location}</h4>
    <h3>{districtData}</h3>
    </div>
  )
}

export default Card;
