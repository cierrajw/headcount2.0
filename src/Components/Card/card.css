.card-div{
  border: 1px solid blue;
  width: 14em;
  height: 30em;
  margin: 0 auto;
  word-wrap: break-word;
  text-align: center;
  padding: 1em;
  margin-bottom: 2em;

}
