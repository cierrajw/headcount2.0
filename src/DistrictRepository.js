

class DistrictRepository{
  constructor(stats){
    super();
    this.stats = {}
  }

  // findByName()
  //
  // findAllMatching()
}

export default DistrictRepository;
