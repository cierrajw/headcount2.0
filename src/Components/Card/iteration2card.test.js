import React from 'react';
import { mount, shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import Card from './Card'
import DistrictRepository from '../../helper.js';
import kinderData from '../../data/kindergartners_in_full_day_program.js';

describe('Card', () =>  {

  const mockData = {
      location: 'Turing',
      stats: {
        2018 : 1
      }
    }

  const district = new DistrictRepository(kinderData);

  it('should match the snapshot', () => {

    const wrapper = shallow(<Card {...mockData}/>);
    expect(wrapper).toMatchSnapshot();
  });
  //
  // it('calls removeIdea with the correct id', () => {
  //
  //   const wrapper = shallow(<Card {...mockData}/>);
  //   removeIdeaMock = jest.fn();
  //
  //
  // });



  // it('should render an h1, controls, and trivialist component', () => {
  //   expect(wrapper.find('h1').length).toEqual(1);
  //   expect(wrapper.find('Controls').length).toEqual(1);
  //   expect(wrapper.find('TriviaList').length).toEqual(1);
  // })

});
