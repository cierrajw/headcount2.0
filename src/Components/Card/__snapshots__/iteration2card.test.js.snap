// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    location="Turing"
    stats={
      Object {
        "2018": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4>
          Turing
        </h4>,
        <h3>
          <p>
            2018
            : 
            1
          </p>
        </h3>,
      ],
      "className": "card-div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Turing",
        },
        "ref": null,
        "rendered": "Turing",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              2018
              : 
              1
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2018",
                ": ",
                1,
              ],
            },
            "ref": null,
            "rendered": Array [
              "2018",
              ": ",
              1,
            ],
            "type": "p",
          },
        ],
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            Turing
          </h4>,
          <h3>
            <p>
              2018
              : 
              1
            </p>
          </h3>,
        ],
        "className": "card-div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Turing",
          },
          "ref": null,
          "rendered": "Turing",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                2018
                : 
                1
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2018",
                  ": ",
                  1,
                ],
              },
              "ref": null,
              "rendered": Array [
                "2018",
                ": ",
                1,
              ],
              "type": "p",
            },
          ],
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    location="Turing"
    stats={
      Object {
        "2018": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4>
          Turing
        </h4>,
        <h3>
          <p>
            2018
            : 
            1
          </p>
        </h3>,
      ],
      "className": "card-div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Turing",
        },
        "ref": null,
        "rendered": "Turing",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              2018
              : 
              1
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2018",
                ": ",
                1,
              ],
            },
            "ref": null,
            "rendered": Array [
              "2018",
              ": ",
              1,
            ],
            "type": "p",
          },
        ],
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            Turing
          </h4>,
          <h3>
            <p>
              2018
              : 
              1
            </p>
          </h3>,
        ],
        "className": "card-div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Turing",
          },
          "ref": null,
          "rendered": "Turing",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                2018
                : 
                1
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2018",
                  ": ",
                  1,
                ],
              },
              "ref": null,
              "rendered": Array [
                "2018",
                ": ",
                1,
              ],
              "type": "p",
            },
          ],
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
